src/main/java/net/mcreator/blockly/data/BlocklyLoader.java

```diff
@@ -26,41 +28,66 @@ public static void init() {
 		INSTANCE = new BlocklyLoader();
 	}
 
-	private final ExternalBlockLoader procedureBlockLoader;
-	private final ExternalBlockLoader jsonTriggerLoader;
-	private final ExternalBlockLoader aitaskBlockLoader;
-	private final ExternalBlockLoader cmdargsBlockLoader;
-	private final ExternalBlockLoader featureBlockLoader;
+	private static final List<String> builtinCategories = new ArrayList<>() {{
+		add("other");
+		add("apis");
+		add("mcelements");
+		add("mcvariables");
+		add("customvariables");
+		add("logicloops");
+		add("logicoperations");
+		add("math");
+		add("text");
+		add("time");
+		add("advanced");
+		add("action");
+		add("aiadvanced");
+	}};
+	private final Map<String, ExternalBlockLoader> blockLoaders;
 	private final ExternalTriggerLoader externalTriggerLoader;
 
 	private BlocklyLoader() {
-		procedureBlockLoader = new ExternalBlockLoader("procedures");
-		aitaskBlockLoader = new ExternalBlockLoader("aitasks");
-		cmdargsBlockLoader = new ExternalBlockLoader("cmdargs");
+		blockLoaders = new LinkedHashMap<>();
+		addBlockLoader("procedures");
+		addBlockLoader("aitasks");
+		addBlockLoader("cmdargs");
+		addBlockLoader("jsontriggers");
+		addBlockLoader("features");
 		externalTriggerLoader = new ExternalTriggerLoader("triggers");
-		jsonTriggerLoader = new ExternalBlockLoader("jsontriggers");
-		featureBlockLoader = new ExternalBlockLoader("features");
 	}
 
-	public ExternalBlockLoader getProcedureBlockLoader() {
-		return procedureBlockLoader;
+	/**
+	 * Create a new {@link ExternalBlockLoader} to load blocks for a specific Blockly panel.
+	 *
+	 * @param name The folder's name where to load files
+	 */
+	public void addBlockLoader(String name) {
+		blockLoaders.put(name, new ExternalBlockLoader(name));
 	}
 
-	public ExternalBlockLoader getAITaskBlockLoader() {
-		return aitaskBlockLoader;
+	/**
+	 * Add a usable category for JSON file blocks that has been created inside the custom XML {@link ToolboxType} file.
+	 * All custom categories used by a {@link ToolboxType} have to be added before blocks are loaded.
+	 *
+	 * @param name The category's name as written inside the XML file (e.g. <i>&lt;custom-thisName/&gt;</i>).
+	 */
+	public static void addBuiltinCategory(String name) {
+		builtinCategories.add(name);
 	}
 
-	public ExternalBlockLoader getCmdArgsBlockLoader() {return cmdargsBlockLoader;}
+	public Map<String, ExternalBlockLoader> getBlockLoaders() {
+		return blockLoaders;
+	}
 
-	public ExternalTriggerLoader getExternalTriggerLoader() {
-		return externalTriggerLoader;
+	public static List<String> getBuiltinCategories() {
+		return builtinCategories;
 	}
 
-	public ExternalBlockLoader getJSONTriggerLoader() {
-		return jsonTriggerLoader;
+	public ExternalBlockLoader getSpecificBlockLoader(String name) {

```

Comment:
I think you can leave out specific here

Reply:
To allow custom Blockly panels in custom mod elements, we still need this new method, so Java plugins can also implement their own Blockly panels (which is the goal of this PR).

Reply:
I suppose Klemen meant the word `Specific` in the method name.

Reply:
> I suppose Klemen meant the word `Specific` in the method name.

Oh ok, my bad.

Reply:
Yep, just meant the name

Reply:
I used Specific to make it more different with the global getter returning all block loaders (and it's called getBlockLoaders)

Reply:
Maybe call the method `getAllBlockLoaders` and then this one `getBlockLoader` then